### PR TITLE
DOC: stats: reorganize hypothesis tests in main page

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
https://github.com/scipy/scipy/pull/9062

#### What does this implement/fix?
This organizes the hypothesis tests section of the stats TOC. 

#### What does this implement/fix?
I'm leaving other sections alone for now (except for moving items into and out of the hypothesis tests section). Other sections can be reconsidered I'm follow-up PRs.

Can items appear in more than one place? For instance, it would be nice to keep the existing "contingency table" section but also list some of those functions as hypothesis tests.

Some functions return confidence intervals but not p-values. Usually, there is an associated hypothesis test that could be added to these functions. In the meantime, it's arguable that these functions belong in the "summary statistics" section.

It would be a nice follow-up to standardize the phrasing of the one-line test descriptions to include distinguishing information. That way, the TOC could almost serve as a decision tree for finding a test appropriate to the analysis to be performed.

### CODE REVIEW

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:240]: `rankdata` was in hypothesis tests. It is *used* in hypothesis tests, but so is `find_repeats`, and neither *are* hypothesis tests.
I think it's reasonable to keep it close to `find_repeats`, wherever that belongs.

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:349]: None of the contingency functions really belong here. Maybe we should leave them where they were, and just move the contingency table hypothesis tests here?

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:320]: There is probably a better way to write this that is more respectful of the difference between "association" and "correlation"?

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:419]: Deprecated.

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:421]: These two *should* be deprecated or marked as legacy, IMO.

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:418]: Once we have `fdr_control`, we can include it with `combine_pvalues` in a Multiple Hypothesis/ Meta-analysis section.

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:417]: Alias for both one- and two- sample tests.

**Review by mdhaber — COMMENTED** [review]
Some self-review.

File: scipy/stats/__init__.py
Comment (tupui) [reply scipy/stats/__init__.py:349]: I would either have a contingency section (and in this section I would separate helper functions from actual functions). Or not respect this distinction and put each method in "functional" sections.

File: scipy/stats/__init__.py
Comment (tupui) [reply scipy/stats/__init__.py:419]: For deprecated things, I think we should have a dedicated section. I don't like mixing things unless there is like a big warning in the description directly in the API listing.

**Review by tupui — COMMENTED** [review]
Thank you for doing this. It's really cleaner 👍 Some point to discuss 😃 

I would not duplicate functions though. This is supposed to be the API doc so we should not add tons of text, duplications, etc. That would be more appropriate for more "narrative"-like docs.

I like the idea to clean the descriptions. About the decision tree, it could actually be good to have one? There is an open PR in `constants` actually to have an interactive svg linking to respective functions. Maybe something to explore. (Although I am not sure that SVG would be the good tech for that). I am also thinking about a stats CLI... Wild ideas there.

File: scipy/stats/__init__.py
Comment (tupui) [inline scipy/stats/__init__.py:444]: I would put these 3 with the other contingency functions. For `crosstab, expected_freq, margins` these are helper functions and can be set aside. Or in the same section but with a line saying these are utilities.

Comment (tupui) [reply scipy/stats/__init__.py:444]: Oh this is what you want to do with the contingency table section you added L443.

Mmm but I think that does not work as in a section named like this I would assume to find the "star" methods.

**Comment by mdhaber** [comment]
> About the decision tree, it could actually be good to have one?

Maybe. I think I would prefer that we link to a reference rather than maintaining our own.

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:372]: I'm not finding a place for `kstest` in this structure, and I think the root cause is discussed in gh-9485.

File: scipy/stats/__init__.py
Comment (mdhaber) [reply scipy/stats/__init__.py:444]: I don't understand.

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:441]: The intent was for this to render like QMC:

image

But it's like:
image

(and so is mstats). Would changing `:maxdepth:` fix this?

File: scipy/stats/__init__.py
Comment (tupui) [reply scipy/stats/__init__.py:444]: What I mean is to do the following grouping.

```
Here are some contingency table functions...

.. autosummary::
    :toctree: generated/

    chi2_contingency
    fisher_exact
    barnard_exact
    boschloo_exact
    contingency.relative_risk
    contingency.association
    contingency.odds_ratio

And here some helper (could be a subsection)

.. autosummary::
    :toctree: generated/

    contingency.crosstab
    contingency.expected_freq
    contingency.margins

```

File: scipy/stats/__init__.py
Comment (tupui) [reply scipy/stats/__init__.py:441]: So we could do that if we add Boschloo and Barnard to this namespace as well.

It's not really by playing with this param, but more by having a proper init in the contingency submodule. This is what we did for QMC. You find the following in its init.

```
Engines
-------

.. autosummary::
   :toctree: generated/

   QMCEngine
   Sobol
   Halton
   LatinHypercube
   PoissonDisk
   MultinomialQMC
   MultivariateNormalQMC

Helpers
-------

.. autosummary::
   :toctree: generated/

   discrepancy
   update_discrepancy
   scale
```

**Review by tupui — APPROVED** [review]
I had a look at the rendered version. It looks good and I would be fine with this version if you don't want to change the contingency part.

@chrisb83 what do you think?

**Comment by mdhaber** [comment]
Yeah I didn't really want mess with contingency. It actually doesn't include any hypothesis tests so I'm fine leaving it as a separate section. But yeah, let's wait for some other feedback here.

**Comment by chrisb83** [comment]
> Can items appear in more than one place? For instance, it would be nice to keep the existing "contingency table" section but also list some of those functions as hypothesis tests.

I do not see a problem to have duplicated items. For example, kstest is a one- and two-sample test, so one could mention it next to ks_2samp and ks_1samp.

**Comment by chrisb83** [comment]
I am fine with the changes, this is clearly an improvement. I was thinking if one should sort the names alphabetically since the order seems a bit random now. i leave it up to you

**Comment by mdhaber** [comment]
The order is not random, but it is subjective. E.g. 

<details>

ttest_1samp and binomtest test specific properties of a distribution. 

```
   skewtest
   kurtosistest
   normaltest
   jarque_bera
   shapiro
```
are all normality tests.

```
   anderson
   cramervonmises
   ks_1samp
   goodness_of_fit
```
are all more general.

```
   chisquare
   power_divergence
```
Are for discrete distributions. They could go at the top, maybe, followed by `binomtest`? I don't know.
</details>

It's not perfect or super careful, but there was some thought.

**Comment by mdhaber** [comment]
Hmm, we've managed to do OK without duplicates except for `kstest`. Given the conclusion in gh-9485, would it be OK to list `kstest` it in the legacy section? That would also indicate that we don't plan to pursue merging additional tests by statistic/null distribution.

**Comment by chrisb83** [comment]
> would it be OK to list kstest it in the legacy section?

Ok

I need to read more about ranksums/tiecorrect

**Comment by mdhaber** [comment]
`ranksums` is the same tests. It just doesn't have an option for continuity correction, and returns the z-score instead of the rank sum statistic. We can certainly add the z-score to mannwhitneyu if desired, and you can get the same p-value with mannwhitneyu with `use_continuity=False`.

`tiecorrect` is a helper function for these tests. I am not aware of any uses outside of the rank sum tests we have.

File: scipy/stats/__init__.py
Comment (mdhaber) [reply scipy/stats/__init__.py:444]: Ah, yes, splitting `contingency` into sections would be a good follow-up, since it doesn't render very well.

Note that only the ones that are not in`contingency` (chi2_contingency, fisher_exact, barnard_exact, boschloo_exact) are actually hypothesis tests. So here, I've just moved those into the hypothesis test section.

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:240]: `tiecorrect` computes something given ranks from `rankdata`, so this is a reasonable place.
```suggestion
   rankdata
   tiecorrect
```

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:424]:
```suggestion
```
I will pursue getting these deprecated, but I'll move them elsewhere for now so we can merge this.

File: scipy/stats/__init__.py
Comment (mdhaber) [inline scipy/stats/__init__.py:365]:
```suggestion
   mannwhitneyu
   ranksums
```

**Review by mdhaber — COMMENTED** [review]
This is at risk of developing nasty merge conflicts, so rather than waiting on an OK to deprecate `ranksums` and `tiecorrect`, I'm moving those into other sections. Hopefully that makes this good to go.

**Review by tupui — APPROVED** [review]
LGTM as a first iteration 👍 @chrisb83 do you want to make further changes or we can integrate this and build on top of that?

**Pull request merged**
